test/utility/test_util.i

```diff
@@ -134,3 +134,44 @@ testutil_make_work_dir(char *dir)
 		    MKDIR_COMMAND, dir);
 	free(buffer);
 }
+
+#undef	M_W
+#define	M_W	(rnd)[0]
+#undef	M_Z
+#define	M_Z	(rnd)[1]
+
+/*
+ * testutil_random_init --
+ *	Initialize return of a 32-bit pseudo-random number.
+ */
+static inline void
+testutil_random_init(uint32_t *rnd)
+{
+	M_W = 521288629;
+	M_Z = 362436069;
+}
+
+/*
+ * testutil_random --
+ *	Return a 32-bit pseudo-random number.
+ *
+ * This is an implementation of George Marsaglia's multiply-with-carry pseudo-
+ * random number generator.  Computationally fast, with reasonable randomness
+ * properties.
+ *
+ * We have to be very careful about races here.  Multiple threads can call
+ * __wt_random concurrently, and it is okay if those concurrent calls get the
+ * same return value.  What is *not* okay is if reading the shared state races
+ * with an update and uses two different values for m_w or m_z.  That could
+ * result in a value of zero, in which case they would be stuck on zero
+ * forever.  Take local copies of the shared values to avoid this.
```

Comment:
You are worried about the two reads being non-atomic, but does it matter if the two writes are are non-atomic? You could force the reads and writes to be atomic by treating `uint32_t rnd[2]` as a 64-bit value which x64 always reads, and writes atomically.